src/commands/gate/evaluate.ts

```diff
@@ -177,27 +177,31 @@ export class GateEvaluateCommand extends Command {
     attempt?: number
   ): Promise<AxiosResponse<EvaluationResponsePayload>> {
     const timePassed = new Date().getTime() - evaluateRequest.startTimeMs
-    const remainingWait = Math.max(0, this.timeoutInSeconds * 1000 - timePassed)
+    const remainingWait = this.timeoutInSeconds * 1000 - timePassed
 
     return new Promise((resolve, reject) => {
-      api
-        .evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout))
-        .then((response) => {
-          if (response.data.data.attributes.status === 'wait') {
-            this.context.stdout.write(renderWaiting())
-            const waitTime = response.data.data.attributes.metadata?.wait_time_ms ?? 0
+      if (remainingWait <= 0) {
+        reject(new Error('wait'))
```

Comment:
if `remainingWait <= 0` we want to stop trying, right? Does it make sense to reject with `'wait'` here?

Reply:
you're right, I'm going to do it in the next PR!